File: server/ruby/send-template-to-list.rb
require 'dotenv'
require 'trycourier'

env_path = File.join(File.dirname(__FILE__), '..', '.env')
Dotenv.load(env_path)

api_key = ENV['COURIER_API_KEY']
list_id = ENV['COURIER_SEND_TEMPLATE_TO_LIST_LIST_ID']
template_id = ENV['COURIER_SEND_TEMPLATE_TO_LIST_TEMPLATE_ID']

client = Trycourier::Client.new(api_key: api_key)

response = client.send_.message(
  message: {
    to: {
      list_id: list_id
    },
    template: template_id
  }
)

puts response

File: server/go/send-template-to-list.go
//go:build ignore

// +build ignore

package main

import (
	"context"
	"fmt"
	"os"
	"path/filepath"

	"github.com/joho/godotenv"
	"github.com/trycourier/courier-go/v4"
	"github.com/trycourier/courier-go/v4/option"
)

func LoadEnv() {
	envPath := filepath.Join("..", ".env")
	godotenv.Load(envPath)
}

func main() {
	LoadEnv()

	apiKey := os.Getenv("COURIER_API_KEY")
	listID := os.Getenv("COURIER_SEND_TEMPLATE_TO_LIST_LIST_ID")
	templateID := os.Getenv("COURIER_SEND_TEMPLATE_TO_LIST_TEMPLATE_ID")

	client := courier.NewClient(
		option.WithAPIKey(apiKey),
	)

	requestBody := map[string]interface{}{
		"message": map[string]interface{}{
			"to": map[string]interface{}{
				"list_id": listID,
			},
			"template": templateID,
		},
	}

	var response map[string]interface{}
	err := client.Post(
		context.Background(),
		"/send",
		requestBody,
		&response,
	)

	if err != nil {
		panic(err.Error())
	}

	fmt.Printf("%+v\n", response)
}
